_posts/2020-03-29-all-you-need-is-lambda-1-booleans.md

```diff
@@ -0,0 +1,260 @@
+---
+layout: post
+title:  "All you need is λ, part one: booleans"
+date:   2020-03-29 20:17:26
+categories: ["lambda calculus", "plt"]
+---
+
+# All you need is λ, part one: booleans
+
+Nearly a hundred years ago, Alonzo Church invented computation in the form of the simple, elegant, and utterly inscrutable lambda calculus. Along with Alan Turing, he then proved that anything computable can be computed in the lambda calculus. However, nearly as soon as we had digital computers, we started inventing programming languages, and with them a vast treasure of features, beautiful and terrible, many of which seem very hard to relate to the fundamental nature of computability, let alone the lambda calculus.
```

Comment:
Rather than

> Along with Alan Turing…

which could imply that they worked together on the lambda calculus, which isn't quite true, I would say

> Just as Alan Turing proved that anything computable can be computed with a Turing machine, so did Church with the lambda calculus…

or something along those lines. You could mention the Church-Turing thesis as well, if you wanted.

Reply:
The proof that the lambda calculus is Turing-complete—Church-Turing—is precisely what that sentence is describing.

I stand by my description of the lambda calculus as utterly inscrutable; in the absence of a metalanguage, it’s like reading binary machine code or something.

Reply:
Does the Church-Turing thesis technically have a proof, since it's an informal statement? Anyway, I thought it would be nice to call it out by name, but your call.

And fair enough. It cannot be said to leap off the page, though I'd say it's easier than x86 assembly. 😆

Reply:
maybe `elusive` instead of `utterly inscrutable`? I do think unfamiliarity with notation makes it challenging to decipher.

Reply:
```
Along with Alan Turing, he then proved that anything computable can be computed in the lambda calculus.
```

Fwiw I also think the section would benefit from a few rounds of refinement. I found the redundancy of the `compute` root in this sentence distracting.